executor/insert_test.go

```diff
@@ -336,6 +336,18 @@ func (s *testSuite3) TestInsertWrongValueForField(c *C) {
 	tk.MustExec(`create table t (a year);`)
 	_, err = tk.Exec(`insert into t values(2156);`)
 	c.Assert(err.Error(), Equals, `[types:8033]invalid year`)
+
+	tk.MustExec(`DROP TABLE IF EXISTS ts`)
+	tk.MustExec(`CREATE TABLE ts (id int DEFAULT NULL, time1 TIMESTAMP NULL DEFAULT NULL)`)
+	tk.MustExec(`SET @@sql_mode=''`)
+	tk.MustExec(`INSERT INTO ts (id, time1) VALUES (1, TIMESTAMP '1018-12-23 00:00:00')`)
+	tk.MustQuery(`SHOW WARNINGS`).Check(testkit.Rows(`Warning 1292 Incorrect timestamp value: '1018-12-23 00:00:00'`))
```

Comment:
In my MySQL8.0, I executed the above statements, it returned the warning of `Out of range`. That's not consistent with the tests.
```
mysql> SET @@sql_mode='';
Query OK, 0 rows affected (0.01 sec)

mysql> CREATE TABLE ts (id int DEFAULT NULL, time1 TIMESTAMP NULL DEFAULT NULL);
Query OK, 0 rows affected (0.07 sec)

mysql> INSERT INTO ts (id, time1) VALUES (1, TIMESTAMP '1018-12-23 00:00:00') ;
Query OK, 1 row affected, 1 warning (0.02 sec)

mysql> show warnings;
+---------+------+------------------------------------------------+
| Level   | Code | Message                                        |
+---------+------+------------------------------------------------+
| Warning | 1264 | Out of range value for column 'time1' at row 1 |
+---------+------+------------------------------------------------+
1 row in set (0.01 sec)

mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.25    |
+-----------+
1 row in set (0.01 sec)
```

Reply:
Yes, TiDB handling of out-of-range differs from MySQL.
Here is an example for `int` instead of `timestamp`:
TiDB:
```sql
tidb> CREATE TABLE t1 (
  id int NOT NULL PRIMARY KEY,
  ts timestamp NULL DEFAULT NULL
);
tidb> INSERT INTO t1 (id, ts) VALUES (2147483648, TIMESTAMP '2018-12-23 00:00:00');
Query OK, 1 row affected, 1 warning (0,00 sec)
tidb> SHOW WARNINGS;
+---------+------+-----------------------------------+
| Level   | Code | Message                           |
+---------+------+-----------------------------------+
| Warning | 1690 | constant 2147483648 overflows int |
+---------+------+-----------------------------------+
1 row in set (0,00 sec)
```
MySQL:
```sql
mysql> INSERT INTO t1 (id, ts) VALUES (2147483648, TIMESTAMP '2018-12-23 00:00:00');
Query OK, 1 row affected, 1 warning (0,01 sec)
mysql> SHOW WARNINGS
+---------+------+---------------------------------------------+
| Level   | Code | Message                                     |
+---------+------+---------------------------------------------+
| Warning | 1264 | Out of range value for column 'id' at row 1 |
+---------+------+---------------------------------------------+
```

I think fixing these differences is outside of the scope for this bug. Do you want me to create a new issue for the mysql incompatibility?

Reply:
I think we can file an issue first.

Reply:
Issue created: https://github.com/pingcap/tidb/issues/26750